src/core/connection.h

```diff
@@ -43,6 +43,11 @@ typedef union QUIC_CONNECTION_STATE {
         BOOLEAN Partitioned     : 1;    // The connection cannot move across partitions.
         BOOLEAN CloseAsync      : 1;    // The connection will close without waiting for callbacks.
 
+        //
+        // The connection's ref count has hit zero, but needs to be cleaned up on a worker.
+        //
+        BOOLEAN CleanupStarted  : 1;
```

Comment:
nit: `Cleanup` feels slightly unclear when we have many steps to end a connection (close / shutdown / free). Maybe `FreePending` would not allow confusion?

Reply:
I prefer `CleanupStarted`: its ambiguity is a feature, not a bug. `FreePending` is something that is either always true for any allocated thing, in a general interpretation, or else is falsely precise, in that the code setting this expects more work to be done than calling `free`.